docker_ci_deploy/__main__.py

```diff
@@ -125,27 +125,32 @@ def run(self, image, tags=None, login=None, registry=None):
         :param registry:
             The address to the Docker registry host.
         """
-        # Build list of tags to push with provided tags
-        push_tags = []
-        if tags is not None:
-            stripped_tag = strip_image_tag(image)
-            for tag in tags:
-                push_tags.append('%s:%s' % (stripped_tag, tag,))
-        else:
-            push_tags = [image]
+        # Build map of images to tags to push with provided tags
+        image_to_push_tags = []
+        for image in images:
+            push_tags = []
+            image_to_push_tags.append((image, push_tags))
+            if tags is not None:
+                stripped_tag = strip_image_tag(image)
+                for tag in tags:
+                    push_tags.append('%s:%s' % (stripped_tag, tag,))
+            else:
+                push_tags.append(image)
 
         # Update tags with registry host information
         if registry is not None:
-            for i, tag in enumerate(push_tags):
-                stripped_tag = strip_image_registry(tag)
-                push_tags[i] = '%s/%s' % (registry, stripped_tag,)
+            for _, push_tags in image_to_push_tags:
+                for i, tag in enumerate(push_tags):
+                    stripped_tag = strip_image_registry(tag)
+                    push_tags[i] = '%s/%s' % (registry, stripped_tag,)
```

Comment:
I don't really like this mutation here. I'd be happier if this were listcomp or something over the values:
`push_tags = ['%s/%s' % (registry, strip_image_registry(tag)) for tag in push_tags]`, perhaps wrapped in a function with a similar thing wrapped around it?

If that's too messy, don't worry about it for now. The mutation was in the existing code already.